daemon/graphdriver/windows/windows.go

```diff
@@ -331,7 +332,19 @@ func (d *Driver) Remove(id string) error {
 	tmpID := fmt.Sprintf("%s-removing", rID)
 	tmpLayerPath := filepath.Join(d.info.HomeDir, tmpID)
 	if err := os.Rename(layerPath, tmpLayerPath); err != nil && !os.IsNotExist(err) {
-		return err
+		// If permission denied, it's possible that the scratch is still mounted, an
+		// artifact after a hard daemon crash for example. Worth a shot to try detaching it
+		// before retrying the rename.
+		if os.IsPermission(err) {
+			if detachErr := vhd.DetachVhd(filepath.Join(layerPath, "sandbox.vhdx")); detachErr != nil {
+				return err
+			}
+			if err = os.Rename(layerPath, tmpLayerPath); err != nil && !os.IsNotExist(err) {
+				return err
+			}
+		} else {
```

Comment:
Nit, but a little cleaner if we return early when "!os.IsPermission(err)"

Reply:
Oooh, missed that comment @cpuguy83. Yes, that's a lot cleaner. Updated.